Index: makefile.win
===================================================================
RCS file: /cvsroot/mozilla/makefile.win,v
retrieving revision 3.106
diff -u -r3.106 makefile.win
--- makefile.win	17 Aug 2001 03:12:26 -0000	3.106
+++ makefile.win	4 Dec 2001 15:25:19 -0000
@@ -152,7 +152,8 @@
 !if defined(FIND)
 !if "$(REBASE)" != ""
 	echo rebasing $(DIST:\=/)
-        $(FIND) $(DIST:\=/) -name "*.dll" > rebase.lst
+#        $(FIND) $(DIST:\=/) -name "*.dll" > rebase.lst
+	dir /s /b $(DIST)\*.dll > rebase.lst
         $(REBASE) -b 60000000 -R . -G rebase.lst
         del rebase.lst
 !else
@@ -166,7 +167,7 @@
 splitsymbols:
 !if defined(MOZILLA_OFFICIAL)
 !if defined(FIND)
-!if "$(SPLITSYM)" != ""
+!if exist($(SPLITSYM))
 	echo splitting symbols out of binaries
 	cd $(MOZ_SRC)\mozilla
 	$(FIND) $(DIST:\=/) -name "*.dll" -exec $(SPLITSYM) {} ;
