diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-atsui.h cairo/src/cairo-atsui.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-atsui.h	2005-05-14 15:20:32.000000000 -0500
+++ cairo/src/cairo-atsui.h	2005-09-05 17:42:40.000000000 -0500
@@ -33,17 +33,17 @@
  *	Calum Robinson <calumr@mac.com>
  */
 
 #ifndef CAIRO_ATSUI_H
 #define CAIRO_ATSUI_H
 
 #include <cairo.h>
 
-#if CAIRO_HAS_ATSUI_FONT
+#ifdef CAIRO_HAS_ATSUI_FONT
 
 /* ATSUI platform-specific font interface */
 
 #include <Carbon/Carbon.h>
 
 CAIRO_BEGIN_DECLS
 
 CAIRO_END_DECLS
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-features.h.in cairo/src/cairo-features.h.in
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-features.h.in	2005-08-12 11:28:37.000000000 -0500
+++ cairo/src/cairo-features.h.in	2005-09-05 17:42:40.000000000 -0500
@@ -46,31 +46,33 @@
 #endif
 
 #define CAIRO_VERSION_MAJOR @CAIRO_VERSION_MAJOR@
 #define CAIRO_VERSION_MINOR @CAIRO_VERSION_MINOR@
 #define CAIRO_VERSION_MICRO @CAIRO_VERSION_MICRO@
 
 #define CAIRO_VERSION_STRING "@CAIRO_VERSION_MAJOR@.@CAIRO_VERSION_MINOR@.@CAIRO_VERSION_MICRO@"
 
-@PS_SURFACE_FEATURE@
+#define @PS_SURFACE_FEATURE@
 
-@PDF_SURFACE_FEATURE@
+#define @PDF_SURFACE_FEATURE@
 
-@XLIB_SURFACE_FEATURE@
+#define @PNG_SURFACE_FEATURE@
 
-@QUARTZ_SURFACE_FEATURE@
+#define @XLIB_SURFACE_FEATURE@
 
-@XCB_SURFACE_FEATURE@
+#define @QUARTZ_SURFACE_FEATURE@
 
-@WIN32_SURFACE_FEATURE@
+#define @XCB_SURFACE_FEATURE@
 
-@GLITZ_SURFACE_FEATURE@
+#define @WIN32_SURFACE_FEATURE@
 
-@FT_FONT_FEATURE@
+#define @GLITZ_SURFACE_FEATURE@
 
-@WIN32_FONT_FEATURE@
+#define @FT_FONT_FEATURE@
 
-@ATSUI_FONT_FEATURE@
+#define @WIN32_FONT_FEATURE@
 
-@PNG_FUNCTIONS_FEATURE@
+#define @ATSUI_FONT_FEATURE@
+
+#define @SANITY_CHECKING_FEATURE@
 
 #endif
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-ft-private.h cairo/src/cairo-ft-private.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-ft-private.h	2005-08-12 11:28:37.000000000 -0500
+++ cairo/src/cairo-ft-private.h	2005-09-05 17:42:40.000000000 -0500
@@ -35,17 +35,17 @@
  */
 
 #ifndef CAIRO_FT_PRIVATE_H
 #define CAIRO_FT_PRIVATE_H
 
 #include <cairo-ft.h>
 #include <cairoint.h>
 
-#if CAIRO_HAS_FT_FONT
+#ifdef CAIRO_HAS_FT_FONT
 
 CAIRO_BEGIN_DECLS
 
 typedef struct _cairo_ft_unscaled_font cairo_ft_unscaled_font_t;
 
 cairo_private cairo_bool_t
 _cairo_unscaled_font_is_ft (cairo_unscaled_font_t *unscaled_font);
 
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-ft.h cairo/src/cairo-ft.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-ft.h	2005-07-21 15:29:34.000000000 -0500
+++ cairo/src/cairo-ft.h	2005-09-05 17:42:40.000000000 -0500
@@ -34,17 +34,17 @@
  *	Owen Taylor <otaylor@redhat.com>
  */
 
 #ifndef CAIRO_FT_H
 #define CAIRO_FT_H
 
 #include <cairo.h>
 
-#if CAIRO_HAS_FT_FONT
+#ifdef CAIRO_HAS_FT_FONT
 
 /* Fontconfig/Freetype platform-specific font interface */
 
 #include <fontconfig/fontconfig.h>
 #include <ft2build.h>
 #include FT_FREETYPE_H
 
 CAIRO_BEGIN_DECLS
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-glitz.h cairo/src/cairo-glitz.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-glitz.h	2005-05-14 15:20:32.000000000 -0500
+++ cairo/src/cairo-glitz.h	2005-09-05 17:42:40.000000000 -0500
@@ -34,17 +34,17 @@
  *	Carl D. Worth <cworth@cworth.org>
  */
 
 #ifndef CAIRO_GLITZ_H
 #define CAIRO_GLITZ_H
 
 #include <cairo.h>
 
-#if CAIRO_HAS_GLITZ_SURFACE
+#ifdef CAIRO_HAS_GLITZ_SURFACE
 
 #include <glitz.h>
 
 CAIRO_BEGIN_DECLS
 
 cairo_surface_t *
 cairo_glitz_surface_create (glitz_surface_t *surface);
 
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-gstate-private.h cairo/src/cairo-gstate-private.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-gstate-private.h	2005-08-24 03:39:28.000000000 -0500
+++ cairo/src/cairo-gstate-private.h	2005-09-05 17:42:40.000000000 -0500
@@ -50,16 +50,18 @@ struct _cairo_gstate {
     cairo_line_join_t line_join;
     double miter_limit;
 
     cairo_fill_rule_t fill_rule;
 
     double *dash;
     int num_dashes;
     double dash_offset;
+    double max_dash_length;
+    double fraction_dash_lit;
 
     cairo_font_face_t *font_face;
     cairo_scaled_font_t *scaled_font;	/* Specific to the current CTM */
     cairo_matrix_t font_matrix;
     cairo_font_options_t font_options;
 
     cairo_clip_t clip;
 
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-pdf.h cairo/src/cairo-pdf.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-pdf.h	2005-05-17 08:17:45.000000000 -0500
+++ cairo/src/cairo-pdf.h	2005-09-05 17:42:40.000000000 -0500
@@ -34,17 +34,17 @@
  *	Carl D. Worth <cworth@cworth.org>
  */
 
 #ifndef CAIRO_PDF_H
 #define CAIRO_PDF_H
 
 #include <cairo.h>
 
-#if CAIRO_HAS_PDF_SURFACE
+#ifdef CAIRO_HAS_PDF_SURFACE
 
 CAIRO_BEGIN_DECLS
 
 cairo_surface_t *
 cairo_pdf_surface_create (const char   *filename,
 			  double	width_in_points,
 			  double	height_in_points);
 
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-ps.h cairo/src/cairo-ps.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-ps.h	2005-05-17 08:17:45.000000000 -0500
+++ cairo/src/cairo-ps.h	2005-09-05 17:42:40.000000000 -0500
@@ -34,17 +34,17 @@
  *	Carl D. Worth <cworth@cworth.org>
  */
 
 #ifndef CAIRO_PS_H
 #define CAIRO_PS_H
 
 #include <cairo.h>
 
-#if CAIRO_HAS_PS_SURFACE
+#ifdef CAIRO_HAS_PS_SURFACE
 
 #include <stdio.h>
 
 CAIRO_BEGIN_DECLS
 
 /* PS-surface functions */
 
 cairo_surface_t *
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-quartz.h cairo/src/cairo-quartz.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-quartz.h	2005-07-21 15:29:34.000000000 -0500
+++ cairo/src/cairo-quartz.h	2005-09-05 17:42:40.000000000 -0500
@@ -34,17 +34,17 @@
  *	Carl D. Worth <cworth@cworth.org>
  */
 
 #ifndef CAIRO_QUARTZ_H
 #define CAIRO_QUARTZ_H
 
 #include <cairo.h>
 
-#if CAIRO_HAS_QUARTZ_SURFACE
+#ifdef CAIRO_HAS_QUARTZ_SURFACE
 
 #include <Carbon/Carbon.h>
 
 CAIRO_BEGIN_DECLS
 
 cairo_surface_t *
 cairo_quartz_surface_create (CGContextRef    context,
 			     int	     width,
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-wideint.h cairo/src/cairo-wideint.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-wideint.h	2005-08-08 17:41:40.000000000 -0500
+++ cairo/src/cairo-wideint.h	2005-09-05 17:42:31.000000000 -0500
@@ -52,17 +52,17 @@
   typedef __int32 int32_t;
   typedef unsigned __int32 uint32_t;
   typedef __int64 int64_t;
   typedef unsigned __int64 uint64_t;
 # ifndef HAVE_UINT64_T
 #  define HAVE_UINT64_T 1
 # endif
 #else
-#error Cannot find definitions for fixed-width integral types (uint8_t, uint32_t, etc.)
+# include "mozstdint.h"
 #endif
 
 /*
  * 64-bit datatypes.  Two separate implementations, one using
  * built-in 64-bit signed/unsigned types another implemented
  * as a pair of 32-bit ints
  */
 
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-win32.h cairo/src/cairo-win32.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-win32.h	2005-05-14 15:20:32.000000000 -0500
+++ cairo/src/cairo-win32.h	2005-09-05 17:43:03.000000000 -0500
@@ -33,17 +33,17 @@
  *	Owen Taylor <otaylor@redhat.com>
  */
 
 #ifndef _CAIRO_WIN32_H_
 #define _CAIRO_WIN32_H_
 
 #include <cairo.h>
 
-#if CAIRO_HAS_WIN32_SURFACE
+#ifdef CAIRO_HAS_WIN32_SURFACE
 
 #include <windows.h>
 
 CAIRO_BEGIN_DECLS
 
 cairo_surface_t *
 cairo_win32_surface_create (HDC hdc);
 
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-xcb-xrender.h cairo/src/cairo-xcb-xrender.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-xcb-xrender.h	2005-05-17 08:11:32.000000000 -0500
+++ cairo/src/cairo-xcb-xrender.h	2005-09-05 17:43:03.000000000 -0500
@@ -34,17 +34,17 @@
  *	Carl D. Worth <cworth@cworth.org>
  */
 
 #ifndef CAIRO_XCB_XRENDER_H
 #define CAIRO_XCB_XRENDER_H
 
 #include <cairo.h>
 
-#if CAIRO_HAS_XCB_SURFACE
+#ifdef CAIRO_HAS_XCB_SURFACE
 
 #include <X11/XCB/xcb.h>
 #include <X11/XCB/render.h>
 
 CAIRO_BEGIN_DECLS
 
 cairo_surface_t *
 cairo_xcb_surface_create_with_xrender_format (XCBConnection	    *c,
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-xcb.h cairo/src/cairo-xcb.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-xcb.h	2005-05-23 17:14:25.000000000 -0500
+++ cairo/src/cairo-xcb.h	2005-09-05 17:43:03.000000000 -0500
@@ -34,17 +34,17 @@
  *	Carl D. Worth <cworth@cworth.org>
  */
 
 #ifndef CAIRO_XCB_H
 #define CAIRO_XCB_H
 
 #include <cairo.h>
 
-#if CAIRO_HAS_XCB_SURFACE
+#ifdef CAIRO_HAS_XCB_SURFACE
 
 #include <X11/XCB/xcb.h>
 
 CAIRO_BEGIN_DECLS
 
 cairo_surface_t *
 cairo_xcb_surface_create (XCBConnection *c,
 			  XCBDRAWABLE	 drawable,
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-xlib-test.h cairo/src/cairo-xlib-test.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-xlib-test.h	2005-05-17 03:06:05.000000000 -0500
+++ cairo/src/cairo-xlib-test.h	2005-09-05 17:43:03.000000000 -0500
@@ -34,17 +34,17 @@
  *	Carl D. Worth <cworth@cworth.org>
  */
 
 #ifndef CAIRO_XLIB_TEST_H
 #define CAIRO_XLIB_TEST_H
 
 #include <cairo.h>
 
-#if CAIRO_HAS_XLIB_SURFACE
+#ifdef CAIRO_HAS_XLIB_SURFACE
 
 #include <cairo-xlib.h>
 
 CAIRO_BEGIN_DECLS
 
 void
 cairo_test_xlib_disable_render (void);
 
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-xlib-xrender.h cairo/src/cairo-xlib-xrender.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-xlib-xrender.h	2005-07-21 01:21:09.000000000 -0500
+++ cairo/src/cairo-xlib-xrender.h	2005-09-05 17:43:03.000000000 -0500
@@ -34,17 +34,17 @@
  *	Carl D. Worth <cworth@cworth.org>
  */
 
 #ifndef CAIRO_XLIB_XRENDER_H
 #define CAIRO_XLIB_XRENDER_H
 
 #include <cairo.h>
 
-#if CAIRO_HAS_XLIB_SURFACE
+#ifdef CAIRO_HAS_XLIB_SURFACE
 
 #include <X11/extensions/Xrender.h>
 
 CAIRO_BEGIN_DECLS
 
 cairo_surface_t *
 cairo_xlib_surface_create_with_xrender_format (Display		 *dpy,
                                                Drawable		  drawable,
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-xlib.h cairo/src/cairo-xlib.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo-xlib.h	2005-07-21 11:52:00.000000000 -0500
+++ cairo/src/cairo-xlib.h	2005-09-05 17:43:03.000000000 -0500
@@ -34,17 +34,17 @@
  *	Carl D. Worth <cworth@cworth.org>
  */
 
 #ifndef CAIRO_XLIB_H
 #define CAIRO_XLIB_H
 
 #include <cairo.h>
 
-#if CAIRO_HAS_XLIB_SURFACE
+#ifdef CAIRO_HAS_XLIB_SURFACE
 
 #include <X11/Xlib.h>
 
 CAIRO_BEGIN_DECLS
 
 cairo_surface_t *
 cairo_xlib_surface_create (Display     *dpy,
 			   Drawable	drawable,
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo.h cairo/src/cairo.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairo.h	2005-08-24 02:21:44.000000000 -0500
+++ cairo/src/cairo.h	2005-09-05 17:43:03.000000000 -0500
@@ -1131,17 +1131,17 @@ void
 cairo_surface_destroy (cairo_surface_t *surface);
 
 cairo_status_t
 cairo_surface_status (cairo_surface_t *surface);
 
 void
 cairo_surface_finish (cairo_surface_t *surface);
 
-#if CAIRO_HAS_PNG_FUNCTIONS
+#ifdef CAIRO_HAS_PNG_FUNCTIONS
 
 cairo_status_t
 cairo_surface_write_to_png (cairo_surface_t	*surface,
 			    const char		*filename);
 
 cairo_status_t
 cairo_surface_write_to_png_stream (cairo_surface_t	*surface,
 				   cairo_write_func_t	write_func,
@@ -1225,17 +1225,17 @@ cairo_image_surface_create_for_data (uns
 				     int			stride);
 
 int
 cairo_image_surface_get_width (cairo_surface_t *surface);
 
 int
 cairo_image_surface_get_height (cairo_surface_t *surface);
 
-#if CAIRO_HAS_PNG_FUNCTIONS
+#ifdef CAIRO_HAS_PNG_FUNCTIONS
 
 cairo_surface_t *
 cairo_image_surface_create_from_png (const char	*filename);
 
 cairo_surface_t *
 cairo_image_surface_create_from_png_stream (cairo_read_func_t	read_func,
 					    void		*closure);
 
diff -rNpU8 /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairoint.h cairo/src/cairoint.h
--- /home/tor/moz/branch/mozilla/gfx/cairo.base/cairo/src/cairoint.h	2005-08-24 03:39:28.000000000 -0500
+++ cairo/src/cairoint.h	2005-09-05 17:44:15.000000000 -0500
@@ -634,29 +634,29 @@ struct _cairo_font_face_backend {
     (*scaled_font_create) (void				*font_face,
 			   const cairo_matrix_t		*font_matrix,
 			   const cairo_matrix_t		*ctm,
 			   const cairo_font_options_t	*options,
 			   cairo_scaled_font_t	       **scaled_font);
 };
 
 /* concrete font backends */
-#if CAIRO_HAS_FT_FONT
+#ifdef CAIRO_HAS_FT_FONT
 
 extern const cairo_private struct _cairo_scaled_font_backend cairo_ft_scaled_font_backend;
 
 #endif
 
-#if CAIRO_HAS_WIN32_FONT
+#ifdef CAIRO_HAS_WIN32_FONT
 
 extern const cairo_private struct _cairo_scaled_font_backend cairo_win32_scaled_font_backend;
 
 #endif
 
-#if CAIRO_HAS_ATSUI_FONT
+#ifdef CAIRO_HAS_ATSUI_FONT
 
 extern const cairo_private struct _cairo_scaled_font_backend cairo_atsui_scaled_font_backend;
 
 #endif
 
 typedef struct _cairo_surface_backend {
     cairo_surface_t *
     (*create_similar)		(void			*surface,
@@ -1010,27 +1010,27 @@ typedef struct _cairo_traps {
 
 #define CAIRO_FONT_SLANT_DEFAULT   CAIRO_FONT_SLANT_NORMAL
 #define CAIRO_FONT_WEIGHT_DEFAULT  CAIRO_FONT_WEIGHT_NORMAL
 
 #define CAIRO_WIN32_FONT_FAMILY_DEFAULT "Arial"
 #define CAIRO_ATSUI_FONT_FAMILY_DEFAULT  "Monaco"
 #define CAIRO_FT_FONT_FAMILY_DEFAULT     ""
 
-#if   CAIRO_HAS_WIN32_FONT
+#if   defined(CAIRO_HAS_WIN32_FONT)
 
 #define CAIRO_FONT_FAMILY_DEFAULT CAIRO_WIN32_FONT_FAMILY_DEFAULT
 #define CAIRO_SCALED_FONT_BACKEND_DEFAULT &cairo_win32_scaled_font_backend
 
-#elif CAIRO_HAS_ATSUI_FONT
+#elif defined(CAIRO_HAS_ATSUI_FONT)
 
 #define CAIRO_FONT_FAMILY_DEFAULT CAIRO_ATSUI_FONT_FAMILY_DEFAULT
 #define CAIRO_SCALED_FONT_BACKEND_DEFAULT &cairo_atsui_scaled_font_backend
 
-#elif CAIRO_HAS_FT_FONT
+#elif defined(CAIRO_HAS_FT_FONT)
 
 #define CAIRO_FONT_FAMILY_DEFAULT CAIRO_FT_FONT_FAMILY_DEFAULT
 #define CAIRO_SCALED_FONT_BACKEND_DEFAULT &cairo_ft_scaled_font_backend
 
 #endif
 
 #define CAIRO_GSTATE_OPERATOR_DEFAULT	CAIRO_OPERATOR_OVER
 #define CAIRO_GSTATE_TOLERANCE_DEFAULT	0.1
